File: Collatz.java
public class Collatz {
    public int solution(int num) {
        int answer = 0;
        long n = num;
        int cnt = 0;

        while(n != 1) {
            if(n % 2 == 0) {
                n = n / 2;
                cnt++;
            }
            else {
                n = (n * 3) + 1;
                cnt++;
            }
            if (cnt >= 500) {
                cnt = -1;
                break;
            }
        }

        answer = cnt;

        return answer;
    }
}

File: Collatz.js
function solution(num) {
    var answer = 0;
    let idx = 0;
    
    while(num != 1) {
        if(num % 2 == 0) {
            num = num / 2;
            idx++;
        }
        else if(num % 2 != 0) {
            num = (num * 3) + 1;
            idx++;
        }
    }
    if(idx >= 500) {
        idx = -1;
    }
    answer = idx;
    return answer;
}